chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -59,8 +59,13 @@ data:
       # the cluster). See clusters option.
       {{- range .Values.clusters }}
       location ~* /api/clusters/{{ .name }} {
+        {{- if .apiServicePath }}
```

Comment:
@absoludity do you know how Helm treats undefined values? It errors, right (null pointer)?

If instead if just passed an empty string or something, I could remove this `if/else` and just rely on:

```
rewrite /api/clusters/{{ .name }}/(.*) /{{ .apiServicePath }}/$1 break;
```
... becomes:
```
rewrite /api/clusters/default/(.*) //$1 break;
``` 

NGINX [merges slashes](https://nginx.org/en/docs/http/ngx_http_core_module.html#merge_slashes) by default.

Reply:
Although adding this extra `apiServicePath` to both the chart and the kubeops codebase will work, I think we can achieve the identical result without it and avoid prematurely introducing extra change and complexity into the `kubeops` codebase. We can do this by using `APIServiceURL` as a URL that can also contain a path - the only reason you were unable to use `APIServiceURL` with a path was because you were trying to pass it verbatim (ie. with the path) through to `proxy_pass`. But we can avoid that in the helm template itself.

So, if we allow `apiServiceURL` to be a URL, which can include a path - such as `https://second-cluster:6443/k8s` then we can actually extract the path from that easily enough in the helm template using [urlParse](https://helm.sh/docs/chart_template_guide/function_list/#urlparse) to pull out the apiServicePath from the resulting dict.

Note that the path will either be an empty string, or a string beginning with a `/`, so you won't need your conditional here, and can instead simply do the following always (note the `/` is no longer needed as it will be in the `apiServicePath`:

```
        rewrite /api/clusters/{{ .name }}/(.*) {{ apiServicePath }}/$1 break;
        rewrite /api/clusters/{{ .name }} {{ apiServicePath }}/ break;
``` 

If you need a hand with this, let us know and we can do an example for you separately. Similarly, when you have the `proxy_pass` command, we can ensure that the `scheme://host:port` only is sent (using the values from urlparse).

With that, I don't think we'll need to change the `kubeops` backend at all, and the existing `apiServiceURL` can define a URL with a path as needed? Let me know what you think or if you have any trouble with that.